WordPress/gutenberg
# `ColorPalette`, `BorderBox`, `BorderBoxControl`: polish and DRY prop types, add default values

## What?


Following up https://github.com/WordPress/gutenberg/pull/45002#pullrequestreview-1145349928 and #44632, this PR refactors the types of the `ColorPalette`, `BorderBox`, and `BorderBoxControl` component in an effort to remove duplicate definitions, fix some wrong/missing details and improve overall code quality.

## Why?


Better types = better docs and better coding experience for the developers consuming the package.

## How?


- `BorderBoxControl` uses types from `BorderControl`, and `BorderControl` uses types from `ColorPalette`. This PR removes some duplicate type defs, and instead re-uses type defs between components
- Added default values for a few props:
  - In a few instances, the docs were mentioning a default value, but that value was not assigned in the code
  - In other instances, the default value assignment was simply moved from the component to the hook file
  - For the remaining instances, those changes are mostly done in order to align the components between themselves and offer more clarity in the docs
- There shouldn't be any runtime changes, apart from the default value assignments 

## Testing Instructions





- Check the components in Storybook, no runtime changes should be noticed compared to `trunk`
- Check that docs, typescript types and README are up to date and in sync with each other

## ✍️ Dev Note

Many props of the `ColorPalette`, `BorderBox` and `BorderBoxControl` components from the `@wordpress/components` package have been assigned a default value which better reflects each prop's purpose.

The related docs have also been updated to reflect those changes.
